### PR TITLE
✨ 🏷️  258 - Keys DAS operations

### PR DESCRIPTION
**Pull request by anncatton**
## Description of Changes

Implementation for Data Mapper Keys DAS service calls AND Keys DAS router endpoints + service calls to support ConsentReleaseData step in the wizard.

### Data Mapper
- adds steps in the main `createConsentReleaseData` function retrieve private keys for ohip and clinicalProfile
- adds supporting service calls for `getOhipKey` and `getClinicalProfileKey` operations. Both these functions are intended to retrieve and existing key OR create and return one if none exists
- ohip flow will only be executed if `hasOhip` is true
- also adds service functions `getOhipKeyByParticipantId` and `getClinicalProfileKeyByParticipantId`. These will only retrieve an existing key, and will return a `<KEY>_NOT_FOUND` error if none exists

### Keys DAS
- adds `upsert` queries to the prisma client to support a `findOrCreate` functionality for the `getOhipKey` and `getClinicalProfileKey` methods (recommended in [prisma docs](https://www.prisma.io/docs/orm/prisma-client/queries/crud#update-or-create-records)
- implements `GET` and `POST` endpoints for both `OhipKeys` and `ClinicalProfileKeys` routers, with validation and error handling
- updates swagger schemas

### Types
- adds `Get` and `Create` req/res types for both `OhipKey` and `ClinicalProfileKey`

## PR Readiness Checklist

- [ ] "Expected Outcome(s)" in ticket have been met
- [x] Ticket number included in PR title
- [x] Connected ticket to PR
- [x] Labels added to PR for service name (`consent-api`, `data-mapper`, etc...), type (`chore`, `documentation`, etc...), status (`draft`, `on-hold`, etc...) **if applicable**
- [x] Manual testing completed
- [x] Builds locally without errors or warnings
- [ ] Tests are updated (if required) and passing
- [ ] PR feedback has been addressed **if applicable**
- [ ] New environment variables added to `.env.schema` files, `README.md`
- [x] Added copyrights to new files

### CODE REVIEW

File: apps/data-mapper/src/services/create.ts [hunk 1]
```diff
@@ -302,7 +310,8 @@ type CreateConsentReleaseDataResponseFailure =
 export const createConsentReleaseData = async (
 	createConsentReleaseDataRequest: ConsentReleaseDataRequest, // this function is intended for the wizard only. the pi update function can be more optional fields to make it more generic
 	participantId: NanoId,
-): Promise<Result<ConsentReleaseDataResponse, CreateConsentReleaseDataResponseFailure>> => {
+	// TODO: decide on required response type
+): Promise<Result<any, CreateConsentReleaseDataResponseFailure>> => {
```
Comment (anncatton) [inline apps/data-mapper/src/services/create.ts:310]: `any` return type fixed in https://github.com/OHCRN/platform/pull/585/files#diff-af22444c08f2624558d67ac1a6c2b533805253ba133f5a0d2ec2d19647457170R321

File: apps/data-mapper/src/services/create.ts [hunk 2]
```diff
@@ -389,14 +400,34 @@ export const createConsentReleaseData = async (
...
 		// fetch or create clinicalProfilePrivateKey
+		const clinicalProfileKeyResponse = await getClinicalProfileKey(participantId);
+		//** PHI DAS service call for ClinicalProfile */
+		if (!isSuccess(clinicalProfileKeyResponse)) {
+			return clinicalProfileKeyResponse;
+		}
 
 		// add clinical data to PHI DAS
+
+		// TODO: decide on required response type
```
Comment (anncatton) [inline apps/data-mapper/src/services/create.ts:425]: response type fixed in https://github.com/OHCRN/platform/pull/585/files#diff-af22444c08f2624558d67ac1a6c2b533805253ba133f5a0d2ec2d19647457170R321 , as mentioned above

File: apps/keys-das/src/prismaClient.ts [hunk 3]
```diff
@@ -47,6 +47,24 @@ const prisma = new PrismaClient().$extends({
...
 			},
+			async upsert({ args, query }) {
+				try {
+					// TODO: is there any scenario where a privateKey is passed in to findOrUpdate?
+					const validPrivateKey = args.create.ohipPrivateKey
+						? NanoId.parse(args.create.ohipPrivateKey)
+						: nanoid();
+					args.create = {
+						...args.create,
+						ohipPrivateKey: validPrivateKey,
```
Comment (anncatton) [inline apps/keys-das/src/prismaClient.ts:58]: update to just set the privateKey to a generated nanoid()

**Review by mistryrn — APPROVED** [review]
Another one with lots of great stuff 🎉 No suggestions, just one question about whether there were any AxiosErrors we might need to account for when doing the upserts 🙂

**Pull request merged**